src/k2/transport/Payload.cpp

```diff
@@ -434,8 +434,9 @@ Payload Payload::shareRegion(size_t startOffset, size_t nbytes){
     return shared;
 }
 
-Payload Payload::copy(BinaryAllocatorFunctor allocator) {
-    Payload copied(allocator);
+Payload Payload::copy(BinaryAllocator allocator) {
```

Comment:
receive by && and then move later on into copied._allocator

Reply:
done, but we need to make a copy of a shared allocator in some cases to avoid making the original allocator empty/invalid.